python/src/spark_rapids_ml/umap.py

```diff
@@ -1259,11 +1423,32 @@ def _construct_umap() -> CumlT:
                 if len(driver_embedding) == 1
                 else np.concatenate([chunk.value for chunk in driver_embedding])
             )
-            raw_data = (
-                driver_raw_data[0].value
-                if len(driver_raw_data) == 1
-                else np.concatenate([chunk.value for chunk in driver_raw_data])
-            )
+
+            if sparse_fit:
+                if not isinstance(driver_raw_data, dict):
+                    raise ValueError("Expected raw data as a CSR dict for sparse fit.")
+                indices = np.concatenate(
+                    [chunk.value for chunk in driver_raw_data["indices"]]
+                )
+                indptr = np.concatenate(
+                    [chunk.value for chunk in driver_raw_data["indptr"]]
+                )
```

Comment:
Can we simply concatenate indptr here? 
Would it be possible that indptr_a = [0, 4], indptr_b=[0, 2, 3] and the indptr_concate should be [0, 4, 6, 7]?

Reply:
Do you mean concatenating on the dictionary value directly? The list comprehension is needed to to retrieve the array from the broadcast object.
Each of the CSR components are one-dim column vectors and are chunked by row; np.concatenate will by default always concatenate by row and the list will preserve the order.

Reply:
Got you. Yeah, need to ensure the values of indptr arrays are not modified by chunking.